src/tests/class_tests/openms/source/File_test.cpp

```diff
@@ -169,6 +169,32 @@ START_SECTION((static bool isDirectory(const String& path)))
   TEST_EQUAL(File::isDirectory(OPENMS_GET_TEST_DATA_PATH("File_test_text.txt")),false)
 END_SECTION
 
+// make source directory and copy it to new location
+// check copy function and if file exists in target path
+START_SECTION(static bool copyDirRecursively(const QString &fromDir, const QString &toDir,File::CopyOptions option = CopyOptions::OVERWRITE))
+  String source_name = OPENMS_GET_TEST_DATA_PATH("XMassFile_test");
+  String target_name = File::getTempDirectory() + "/" + File::getUniqueName() + "/"; 
+  QDir sdir;
+  TEST_EQUAL(sdir.mkpath(source_name.toQString()), true)
+  // test canonical path
+  TEST_EQUAL(File::copyDirRecursively(source_name.toQString(),source_name.toQString()),false)
+  // test copy overwrite default
+  TEST_EQUAL(File::copyDirRecursively(source_name.toQString(),target_name.toQString()),true)
+  TEST_EQUAL(File::exists(target_name + "/pdata/1/proc"),true);
+  File::removeDirRecursively(target_name);
+  // test copy overwrite 
+  TEST_EQUAL(File::copyDirRecursively(source_name.toQString(),target_name.toQString(), File::CopyOptions::OVERWRITE),true)
+  TEST_EQUAL(File::exists(target_name + "/pdata/1/proc"),true);
```

Comment:
mhh.. how do you know the file was actually overwritten? Could have been skipped as well.
You'd have to check the content e.g. by emptying the content first (e.g. ofstream.open()+close()), then overwrite and check content again (or just size).